### PR TITLE
Improve catalog container usability

### PR DESCRIPTION
**Pull request by arbulu89**
# Description
Make the `CatalogContainer` usage a bit more generic. This will help us to reuse this component in other places where the catalog data is queried from wanda.
In order to achieve that, I have added a new reducer, selector and saga combo. This helps on testing side as well.

## How was this tested?

Reducers, selectors and sagas tests added. In order to test the latest one, I have added the `axios-mock-adapter` to mock the api calls.

Here some references on how to test redux code (official docs):
https://redux.js.org/usage/writing-tests
https://redux-saga.js.org/docs/advanced/Testing/

### CODE REVIEW

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Just some changes then we can merge

File: assets/js/state/index.js [hunk 1]
```diff
@@ -28,6 +29,7 @@ export const store = configureStore({
     sapSystemsList: sapSystemListReducer,
     databasesList: databasesListReducer,
     catalog: catalogReducer,
+    catalog_new: catalogNewReducer,
```
[the remark this answers is not in the document]
Comment (arbulu89) [reply assets/js/state/index.js:32]: Yes, once we do the transition to the new wanda engine, we will remove all of this `new` suffixes and remove old catalog code.
PD: Damn camelCase, snake_case and all possible options XD

[the remark this answers is not in the document]
Comment (arbulu89) [reply assets/js/state/index.js:32]: @dottorblaster would it make sense to start putting the reducers code in a `reducers` folder? within the `state` folder

[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/state/index.js:32]: > PD: Damn camelCase, snake_case and all possible options XD

also kebab-case! :smile: 

> would it make sense to start putting the reducers code in a reducers folder?

Maybe, but these are not proper reducers per-se, maybe a `slices` directory? Maybe it's better to have this on the radar but deferring it right now

[the remark this answers is not in the document]
Comment (arbulu89) [reply assets/js/state/index.js:32]: @dottorblaster Forget about it. I have been reading redux guidlines, and they suggest to create folders by logical components. But we are really far from that concept.
We can discuss in the future if we want/can update the folders structure

[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/state/index.js:32]: Yeah we should totally revisit this @arbulu89, I'll create a tech debt to track this, good catch

**Comment by arbulu89** [comment]
Hey @dottorblaster ,
I have applied the requested changes, and even included some improvements to use the new `wanda` api code, so we have a centralized api usage.
I'm ommiting by now the `error/errorMessage` change if you don't mind. We already discussed offline the option to have some sort of `Boolean` selector, but we can include that afterwards

**Review by dottorblaster — APPROVED** [review]
> and even included some improvements to use the new wanda api code, so we have a centralized api usage

I forgot to tell you about that, you did the right thing!

**Pull request merged**